apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnectionStringBuilder.cs

```diff
@@ -41,7 +43,57 @@ public sealed class SpannerConnectionStringBuilder : DbConnectionStringBuilder
         /// Credentials can be retrieved from a file or obtained interactively.
         /// See Google Cloud documentation for more information.
         /// </summary>
-        public ITokenAccess Credential { get; private set; }
+        public ITokenAccess GetCredential()
+        {
+            // Check the ctor override.
+            if (CredentialOverride != null)
+            {
+                return CredentialOverride;
+            }
+
+            //Calculate it from the CredentialFile argument in the connection string.
+            GoogleCredential credential = null;
+            string jsonFile = CredentialFile;
+            if (!string.IsNullOrEmpty(jsonFile))
+            {
+                if (string.Compare(Path.GetExtension(jsonFile), ".json", StringComparison.OrdinalIgnoreCase) != 0)
+                {
+                    throw new ArgumentException($"{nameof(CredentialFile)} should only be set to a JSON file.");
+                }
+
+                if (!File.Exists(jsonFile))
```

Comment:
Is there precedence for this in other ADO.NET providers? For example, SqlCeConnection specifies a file as a data source - does that do this sort of thing? It's one of those things like lenient conversions - useful in some cases, but hard to predict in others. I'm not definitely against it - just slightly wary.

Reply:
I did find an example of another provider supporting a file through a path -- it might have been MySql.CertificateFile which is not a credential, but an ssl pem file IIRC.  I would have liked to support a resource as I think EF creates its own "res:" uri syntax, but that's too much work and duplication of what they do.  I'm hoping once we move to EF that we can support that better.